content/docs/rfcs/35/README.md

```diff
@@ -0,0 +1,336 @@
+
+```abnf
+; protocol ID
+protocol-id                 = 1OCTET
+
+; contains the size of handshake-message
+handshake-message-len       = 1OCTET
+
+; contains one or more Diffie-Hellman public keys
+handshake-message           = *OCTET
+
+; contains the size of message-len
+transport-message-len-len   = 1OCTET
+
+; contains the size of transport-message
+transport-message-len       = *OCTET
+
+; contains the transport message, eventually encrypted
+transport-message           = *OCTET
+
+; contains authentication data for transport-message, if encrypted
+transport-message-auth        = 16OCTET
+
+; the Waku WakuMessage payload field
+payload  = protocol-id handshake-message-len handshake-message transport-message-len-len transport-message-len transport-message transport-message-auth 
+```
+
+### Protocol Payload Format
+
+Based on the specified `protocol-id`, 
+the Waku message `payload` field will encode different types of protocol-dependent messages. 
+
+In particular, if `protocol-id` is
+- `0`:  payload encodes an [after-handshake](#After-handshake) message.  
+    - `handshake-message-len` MAY be 0;
+    -  `transport-message` contains the Noise transport message;
+- `10`,`11`,`12`,`13`: payload encodes a supported Noise handshake message.  
+    -  `transport-message` contains the Noise transport message;
+- `30`: payload encapsulate a `ChaChaPoly` ciphertext `ct`.  
+    - `handshake-message-len` is set to `0`;
+    - `transport-message` contains the concatenation of the encryption nonce (12 bytes) followed by the ciphertext `ct`;
+    - `transport-message-len-len` and `transport-message-len` are set accordingly to `transport-message` length;
+    - `transport-message-auth` contains the authentication data for `ct`.
+
+
+### Public Keys Serialization
+
+Diffie-Hellman public keys can be trasmitted in clear 
+or in encrypted form with authentication data attached. 
+To distinguish between these two cases, public keys are serialized as the concatenation of the following three fields:
+
+- `flag`: 
+is equal to `1` if the public key is encrypted; 
+`0` otherwise (1 byte);
+- `pk`: 
+if `flag = 0`, it contains an encoding of the X coordinate of the public key. 
+If `flag = 1`, it contains a symmetric encryption of an encoding of the X coordinate of the public key;
+- `pk-auth`: 
+if `flag = 0`, it is empty; 
+if `flag = 1`, it contains the authentication data for `pk`;
+
+The corresponding serialization is obtained as `flag pk pk-auth`.
+
+As regards the underlying supported [cryptographic primitives](#Cryptographic-primitives):
+- `Curve25519` public keys X coordinates are encoded in little-endian as 32 bytes arrays;
+- `ChaChaPoly` authentication data consists of 16 bytes 
+(nonces are implicitely defined by Noise [processing rules](http://www.noiseprotocol.org/noise.html#processing-rules)).
+
+In all supported Noise protocols, 
+parties' static keys are transmitted encrypted, 
```

Comment:
```suggestion
parties' static public keys are transmitted encrypted, 
```

Reply:
Addressed in https://github.com/vacp2p/rfc/pull/496/commits/4076b1e0f22d497b0ac6c67727424496f30d34ac